### PR TITLE
Added Sun_checks Checkstyle with custom supression script and other integrations \ Danshal

### PR DESCRIPTION
**Pull request by levilime**
User Story: As a programmer I want to make informed decisions on the quality of a merge request by using Continuous integration.

Because the connector is mostly code not written or maintained by us, we'll only do checkstyle on the code that we touch. In this way we provide strong arguments that we can do checkstyle well (by example) while not running in constantly correct checkstyle mistakes from the repo we fork off. This would be an uphill battle.

The python script on the server is run by using jython. This is because Travis only supports one development language per integration.

Cobertura fails when the total line coverage for the Environment is lower than 75%.

It would be advisible to check the code with checkstyle before commiting. You can run the eclipse plugin with the official Sun style and then only correct the style for the lines you've edited.

PMD and Findbugs are just outputted in the Travis Log.

### CODE REVIEW

**Comment by spiderbiggen** [comment]
This looks good. will be very useful to have the CI only run these checks on our changes. and not on everything

**Comment by JoshCode** [comment]
I suggest that the lines 

```
notifications:
  slack: $slack_token
```

from #4 are added so that when pulling this .travis.yml into our own repositories, Slack integration will still work and this .travis.yml will never have to be edited by other groups anymore, which could result in overwriting things, because different groups that want Travis CI integration will have different tokens, so when pushing commits, the new .travis.yml will get pushed too, resulting in invalid tokens being pulled to other team's repo's.

\- CodeFox [CodeFox](https://github.com/codefoxnl)

**Comment by spiderbiggen** [comment]
@JoshCode what would happen here if you don't add a slack token to your variables on travis? nothing?

**Comment by MaxGroenenboom** [comment]
Should all groups use the same checkstyle rules?

**Comment by JoshCode** [comment]
@spiderbiggen Yeah, nothing should happen. Perhaps a warning, but not a failure.

**Comment by levilime** [comment]
@MaxGroenenboom I think it is logical all groups should use the same checkstyle, because we'll all use the same integration (that on eishub/tygron context) This pull request is also about which checkstyle we should use together. I think google checkstyle is a good one.

**Comment by levilime** [comment]
@JoshCode isn't it possible to integrate Travis with a git repo through the webhooks, and it being connected to your group fork? Is it necessary to have this token there? But I can add it if it is useful to you or others :smile:

**Comment by JoshCode** [comment]
@levilime It is for the Slack - Travis CI integration, which is only possible with a token, or we'd have to write an entire integration bot ourselves ;)

**Comment by MaxGroenenboom** [comment]
As I said during the meeting, we (WhySoSerious) don't agree yet on the choice of the checkstyle format, just leaving it here as well for the record.

**Comment by spiderbiggen** [comment]
@MaxGroenenboom but then what do you as a group suggest

**Comment by MaxGroenenboom** [comment]
What I would suggest as a person (so not speaking for my group right now):

A format that looks a bit more like what the connector is right now. The Google format yields over 2500 warnings, while the Sun Checks (Eclipse) configuration yields only over 500 warnings. This means if we use the Sun Checks (Eclipse) configuration, our added code is much more consistent with the existing code, which is more important than our code looking "nice" according to some random company.

**Comment by levilime** [comment]
@MaxGroenenboom I like your thinking, I think you yield good arguments to change it to Sun checkstyle. I will change it to

**Comment by MaxGroenenboom** [comment]
Also, is there any good reason to use spaces instead of tabs? I know a lot of software engineers prefer using spaces but I never heard a good reason for it, and as the entire connector currently uses tabs instead of spaces, I think it would be better to use tabs for the same reason I wrote in my last comment on this issue.

By the way, Sun Checks has a modified configuration for Eclipse (which is made conform the default Eclipse styling configuration), which is the one I recommended. This configuration by default uses tabs so we could use it? I think most people use Eclipse anyway so I see no reason not to.

**Comment by levilime** [comment]
@MaxGroenenboom OLD The sun checkstyle configuration is now used. It doesn't fail on tabs that I know of OLD. Oops, nope that's the FileTabCharacter property. I think you have nice convincing arguments for no tabs so will take that from the sun checkstyle.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
+[![Build Status](https://travis-ci.org/eishub/tygron.svg?branch=context)](https://travis-ci.org/eishub/tygron)
+
 # Tygron EIS connector
 
-Tygron is an EIS environment that connects with the tygron http://www.tygron.com system for urban planning.
+Tygron is an EIS environment that connects with the tygron http://www.tygron.com sytem fsor urban planning.
```
Comment (JoshCode) [inline README.md]: Typo

**Comment by levilime** [comment]
Please merge :point_right: :red_circle:

**Comment by spiderbiggen** [comment]
This looks good please merge \ Rampestampers

**Comment by Denpeer** [comment]
Looks like it's ready to merge \ WhySoSerious

**Pull request merged**